modules/imgproc/include/opencv2/imgproc/hal/hal.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CV_IMGPROC_HAL_HPP
+#define CV_IMGPROC_HAL_HPP
+
+#include "opencv2/core/cvdef.h"
+#include "opencv2/core/hal/interface.h"
+
+namespace cv { namespace hal {
+
+//! @addtogroup core_hal_functions
+//! @{
+
+struct FilterContext
+{
+    void * impl;
+    FilterContext() : impl(0) {}
+};
+
+
+CV_EXPORTS
+void init_filter2d(FilterContext &c,
```

Comment:
whenever we keep C-style API or move to C++, let's use mixed-case naming style "initFilter2D", like we do everywhere else in the library